terraform/module-prometheus/prometheus.tf

```diff
@@ -71,45 +71,42 @@ resource "aws_security_group_rule" "prometheus_to_himself_instance_metrics" {
 ###
 
 resource "aws_instance" "prometheus" {
-  ami = "${data.aws_ami.debian.id}"
+  ami = data.aws_ami.debian.id
 
   # associate_public_ip_address = false
   count                = 1
-  iam_instance_profile = "${aws_iam_instance_profile.prometheus.name}"
-  instance_type        = "${var.prometheus_type}"
-  key_name             = "${var.keypair_name}"
-  ebs_optimized        = "${var.prometheus_ebs_optimized}"
+  iam_instance_profile = aws_iam_instance_profile.prometheus.name
+  instance_type        = var.prometheus_type
+  key_name             = var.keypair_name
+  ebs_optimized        = var.prometheus_ebs_optimized
 
-  vpc_security_group_ids = ["${compact(list(
-    "${var.bastion_sg_allow}",
-    "${aws_security_group.prometheus.id}",
-  ))}"]
+  vpc_security_group_ids = compact([var.bastion_sg_allow, aws_security_group.prometheus.id])
 
   # subnet_id = "${element(var.private_subnets_ids, count.index)}"
-  subnet_id = "${element(var.public_subnets_ids, count.index)}"
+  subnet_id = element(var.public_subnets_ids, count.index)
```

Comment:
I stlll dont know in this case if the new format is better or not

`subnet_id = var.public_subnets_ids[count.index]`